frmts/grok/grokdataset.cpp

```diff
@@ -0,0 +1,4173 @@
+/******************************************************************************
```

Comment:
Apart from Grok namespacing instead of openjpeg, a side-to-side diff with frmts/openjpeg/openjpegdataset.cpp shows that this file is at > 90% identical to it. We are speaking of 4173 lines for grokdataset.cpp and 4334 lines for openjpegdataset.cpp. To the best of my knowledge, this is a unique situation in the project. I'm not sure what to do in that situation:
- sigh, sigh and sigh again and accept that sad situation
- ask Grok to be relicensed into the original BSD 2 license and rip off the openjpeg driver
- use C++ templating to have one common implementation, or C++ inheritance with virtual methods
- looser variation of the above: extract parts that are independent from the underlying library so they can be reused by both. It looks like JP2 box formatting could be a candidate for that.

Reply:
Latest version now has common code refactored into base `JP2Dataset` class. This moves ~25% of the code into the base class. As the remaining methods have codec-specific code interleaved with more generic code, I do not think it is worth the effort at the moment to split it any further. Of course this can be done in the future if it makes sense. Let me know what you think.